machineconfiguration/v1/types.go

```diff
@@ -0,0 +1,531 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+)
+
+// CustomResourceDefinition for MCOConfig
+// apiVersion: apiextensions.k8s.io/v1beta1
+// kind: CustomResourceDefinition
+// metadata:
+//   # name must match the spec fields below, and be in the form: <plural>.<group>
+//   name: mcoconfigs.machineconfiguration.openshift.io
+// spec:
+//   # group name to use for REST API: /apis/<group>/<version>
+//   group: machineconfiguration.openshift.io
+//   # list of versions supported by this CustomResourceDefinition
+//   versions:
+//     - name: v1
+//       # Each version can be enabled/disabled by Served flag.
+//       served: true
+//       # One and only one version must be marked as the storage version.
+//       storage: true
+//   # either Namespaced or Cluster
+//   scope: Namespaced
+//   names:
+//     # plural name to be used in the URL: /apis/<group>/<version>/<plural>
+//     plural: mcoconfigs
+//     # singular name to be used as an alias on the CLI and for display
+//     singular: mcoconfig
+//     # kind is normally the CamelCased singular type. Your resource manifests use this.
+//     kind: MCOConfig
+//     # shortNames allow shorter string to match your resource on the CLI
+//     shortNames:
+//
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MCOConfig describes configuration for MachineConfigOperator.
+type MCOConfig struct {
```

Comment:
Do we have plans to move to `https://github.com/openshift/api/tree/master/operator/v1` like all the other operators..?

Reply:
cc @runcom

Reply:
@abhinavdahiya this PR is indeed the start of what you're asking or am I missing something? the MachineConfig struct type cannot be moved right away as it depends on ignition and we don't control tags on struct's fields so it's trickier and will be done in the 4.1.x/4.2 timeframe

Reply:
@abhinavdahiya you're just talking about the actual `MCOConfig` struct, so tl;dr; yeah, we'll move that to `operator/v1`

Reply:
will be doen as a followup